controllers/classController.js

```diff
@@ -29,22 +33,32 @@ exports.createClass = async (req, res) => {
     );
 
     if (imageUrl) fs.unlinkSync(image);
+git 
     // Get creator info from DB
     const creatorInfo = await Users.findById({ _id });
+    const reInfo = await resourcePerson.findOne({ userId: _id });
 
     // Create and Save info in DB
     const createClass = await Class.create({
       className,
       description,
       price,
+      duration,
+      curriculum,
+      course: reInfo.course,
       pictureUrl: imageUrl.url,
       creatorId: _id,
       creatorPics: creatorInfo.profile_picture
     });
 
+    // Create and Save info in DB
+    const createCurriculum = await Curriculum.create({
+      curriculum, course: reInfo.course, classid: createClass._id
```

Comment:
Also, are you are getting the course from the course attached to the resource person? 
Please explain @BigB97

Reply:
Like we discussed the other day, we don't know the exact course the resource person is teaching until they tell us the course (e.g MATHEMATICS, ENGLISH, BIOLOGY....), so when this resource person update their respective course, we can use it when creating the class to know
1. This  class belong to MATHEMATICS course.
2. And when displaying list of resource person for student to pick we need to know which course does this class belong to and find resource person who are for that course